front/lib/api/data_sources.ts

```diff
@@ -363,13 +363,24 @@ export async function upsertDocument({
   // Data source operations are performed with our credentials.
   const credentials = dustManagedCredentials();
 
+  const documentId = name;
+  const documentParents = parents || [];
+
+  // Ensure that the documentId is included in the parents as the first item.
+  // remove it if it's already present and add it as the first item.
+  const indexOfDocumentId = documentParents.indexOf(documentId);
+  if (indexOfDocumentId !== -1) {
+    documentParents.splice(indexOfDocumentId, 1);
+  }
+  documentParents.unshift(documentId);
```

Comment:
This one is not guaranteed ImHo

But as a starter we could add a log with assert if we find parents that don't have the documentId in them?

Reply:
I'll drop it for now.

Reply:
This is only called in front from the private api when people upsert documents manually, are you sure ?

Reply:
Sorry got it. Misread it. I thought it was a conenctors generic thing. 

That being said it's not limited to folders is it? All connection upserts will run through that code no?

Reply:
Hum, beyond folder, outside of connectors, what can upsert ? (right now I only see folder usage, soon attachments uploads but it's also folders).

What do you have in mind ?

Reply:
Connectors. They go through this code path right?

Reply:
No (but there is a function with the same name.. can be quite confusing :))

Reply:
kk! Let's maybe rename that function while we're here?

Reply:
Yup.